app/controllers/email_confirmations_controller.rb

```diff
@@ -128,4 +139,9 @@ def login_failure(message)
     flash.now.alert = message
     render template: "multifactor_auths/mfa_prompt", status: :unauthorized
   end
+
+  def session_active?
+    return false if session[:mfa_expires_at].nil?
+    session[:mfa_expires_at] > Time.current
```

Comment:
This might be clearer as an if-else expression, but I feel like I've seen this pattern used before.